File: apps/website/src/services/order.ts
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

import { OrderData } from '../models'
import { appvars } from '../config'
import { CrudService } from './base/crudService'

const OrderService = new CrudService<OrderData>('order', appvars.ENDPOINT.ORDER, {})

export default OrderService

File: apps_opt_engine/core/core-impl/src/main/java/dev/aws/proto/core/Order.java
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package dev.aws.proto.core;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Objects;

@Data
@AllArgsConstructor
public class Order {

    private String orderNo;
    private String orderDate;
    private String deliveryCode;
    private double sumWeight;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Order that = (Order) o;
        return Objects.equals(this.orderNo, that.orderNo);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(this.orderNo);
    }
}
